lua/framework/lua_app.cc

```diff
@@ -222,17 +222,6 @@ LuaApp::RunInteractive(int argc, char** argv)
 int
 LuaApp::RunBatch(int argc, char** argv)
 {
-#ifndef NDEBUG
-  opensn::log.Log() << "Waiting...";
-  if (opensn::mpi_comm.rank() == 0)
-    for (int k = 0; k < 2; ++k)
-    {
-      usleep(1000000);
-      opensn::log.Log() << k;
-    }
-  mpi_comm.barrier();
-#endif
-
```

Comment:
could this be due to the fact that it takes a bit of time for the the debugger (totalview, for instance) to start in parallel?

Reply:
I typically do:
```
mpirun -np <n> ./binary params -- -start_in_debugger
```
the `-start_in_debugger` is a PETSc option that will print the commands how to attach a debugger to the new processes. Then, I open <n> terminals, paste these commands there and run.

But, I think I want to keep this:
```
#ifndef NDEBUG
  mpi_comm.barrier();
#endif
```
so that the processes waiting until all are started.

Reply:
I checked and the barrier is not needed, PETSc does the right thing...